### PR TITLE
Allow SecurityComponent to check fields that are optional, but must have a specific value if included

### PR DESCRIPTION
**Pull request by Zuluru**
See issue #7848 for discussion. This PR supersedes PR #8558, which got messed up due to incorrect git management, and was never checked with v3.3.

I haven't yet made any changes in the documentation, pending acceptance of general implementation of this PR. If these changes look good but you want the docs updated before merging, let me know and I can take care of that.

### CODE REVIEW

File: src/View/Helper/FormHelper.php [hunk 1]
```diff
@@ -601,12 +622,19 @@ public function secure(array $fields = [], array $secureAttributes = [])
             'value' => $tokenData['unlocked'],
         ]);
         $out .= $this->hidden('_Token.unlocked', $tokenUnlocked);
+        if (!empty($tokenData['optional'])) {
+            $tokenOptional = array_merge($secureAttributes, [
```
Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: if those are all string based array keys (assoc. array), it is better to use + operator for joining

    $attributes = ['value' => $tokenData['optional'] + $secureAttributes;

Comment (Zuluru) [inline src/View/Helper/FormHelper.php]: I was following the code examples in the lines before and after, which all use array_merge.

**Comment by thinkingmedia** [comment]
Looks good to me. 👍

File: src/View/Helper/FormHelper.php (hunk 1 above)
Comment (dereuromark) [inline src/View/Helper/FormHelper.php]: Indeed, those are all kind of the inferior operator. But this is then out of scope of this PR and can be cleaned up later.

Comment (Zuluru) [inline src/View/Helper/FormHelper.php]: I'll take care of that if/when this is accepted.

Comment (Zuluru) [inline src/View/Helper/FormHelper.php]: Best to leave my current changes as-is for consistency until then?

**Comment by markstory** [comment]
After some internal discussion with other folks on the team, I am going to close this issue. We had concerns about the complexity this adds and how infrequently optional field usage would be. Thank you for taking the time to put these changes together, but I don't feel comfortable taking on the long term maintenance costs.

**Pull request closed without merging**